pkg/kubectl/cmd/annotation.go

```diff
@@ -0,0 +1,211 @@
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/resource"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/spf13/cobra"
+)
+
+const (
+	annotation_long = `Update the annotations on one or more resources.
+
+An annotation is a key/value pair that can hold larger (compared to a label), and possibly not human-readable, data.
+It is intended to store non-identifying auxiliary data, especially data manipulated by tools and system extensions.
+If --overwrite is true, then existing annotations can be overwritten, otherwise attempting to overwrite an annotation will result in an error.
+If --resource-version is specified, then updates will use this resource version, otherwise the existing resource-version will be used.`
+	annotation_example = `// Update pod 'foo' with the annotation 'source_id' and the value 'foo-v1:7764ae47-9092-11e4-8393-42010af018ff'.
+$ kubectl annotation pods foo source_id=foo-v1:7764ae47-9092-11e4-8393-42010af018ff
+
+// Update pod 'foo' with the annotation 'source_id' and the value 'foo-v1:c70ce5c8-3165-11e5-a151-feff819cdc9f', overwriting any existing value.
+$ kubectl annotation --overwrite pods foo source_id=foo-v1:c70ce5c8-3165-11e5-a151-feff819cdc9f
+
+// Update all pods in the namespace
+$ kubectl annotation pods --all source_id=foo-v1:c70ce5c8-3165-11e5-a151-feff819cdc9f
+
+// Update pod 'foo' only if the resource is unchanged from version 1.
+$ kubectl annotation pods foo source_id=foo-v1:c70ce5c8-3165-11e5-a151-feff819cdc9f --resource-version=1
+
+// Update pod 'foo' by removing an annotation named 'source_id' if it exists.
+// Does not require the --overwrite flag.
+$ kubectl annotation pods foo source_id-`
+)
+
+func NewCmdAnnotation(f *cmdutil.Factory, out io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "annotation [--overwrite] RESOURCE NAME KEY_1=VAL_1 ... KEY_N=VAL_N [--resource-version=version]",
+		Short:   "Update the annotations on a resource",
+		Long:    annotation_long,
+		Example: annotation_example,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := RunAnnotation(f, out, cmd, args)
+			cmdutil.CheckErr(err)
+		},
+	}
+	cmdutil.AddPrinterFlags(cmd)
+	cmd.Flags().Bool("overwrite", false, "If true, allow annotations to be overwritten, otherwise reject annotation updates that overwrite existing annotations.")
+	cmd.Flags().Bool("all", false, "select all resources in the namespace of the specified resource types")
+	cmd.Flags().String("resource-version", "", "If non-empty, the annotation update will only succeed if this is the current resource-version for the object. Only valid when specifying a single resource.")
+	return cmd
+}
+
+// validateNoAnnotationOverwrites validates that when overwrite is false, to-be-updated annotations don't exist in the object annotation map (yet)
+func validateNoAnnotationOverwrites(meta *api.ObjectMeta, annotations map[string]string) error {
```

Comment:
It would be more user-friendly to report all the collisions at once instead of having the user fix them one at a time.  This could also encourage the user to `--overwrite` with the intent of clobbering annotationA, but also end up clobbering annotationB.